services/checkin/controller/controller.go

```diff
@@ -58,12 +63,26 @@ func GetCurrentUserCheckin(w http.ResponseWriter, r *http.Request) {
 }
 
 /*
-	Endpoint to set the specified user's checkin
+Endpoint to set the specified user's checkin
 */
 func CreateUserCheckin(w http.ResponseWriter, r *http.Request) {
 	var user_checkin models.UserCheckin
 	json.NewDecoder(r.Body).Decode(&user_checkin)
 
+	err := validate.Struct(user_checkin)
+	if err != nil {
+		errors.WriteError(w, r, errors.AttributeMismatchError(err.Error(), "No user token was provided."))
+		return
+	}
+
+	id, err := utils.FetchIdFromSignedUserToken(common_config.TOKEN_SECRET, user_checkin.UserToken)
+	if err != nil {
+		errors.WriteError(w, r, errors.AuthorizationError(err.Error(), "Bad user token."))
+		return
+	}
+
+	user_checkin.ID = id
```

Comment:
We can probably remove the id field from here, as it isn't really required.

Reply:
While id is not used on input, it is used on output and when returning to the user